src/shared/db.ts

```diff
@@ -1,3 +1,21 @@
 import { PrismaClient } from "@prisma/client";
+import { AdapterUser } from "next-auth/adapters";
 
 export const prisma = new PrismaClient();
+
+export async function createUser(loginUser: AdapterUser) {
+  const existingUser = await prisma.user.findUnique({
+    where: { email: loginUser.email },
```

Comment:
as the email and the username are unique, you should check both

Reply:
`loginUser` comes from the third-party providers we're using, in this case Google and Github.

As an `AdapterUser`  interface, it doesn't have a `username` property:

![image](https://github.com/podcodar/app/assets/107960652/f7c2cada-e347-4dde-b51a-805879fa80e1)

What we're doing here is checking if the user signing in with 3P provider is already in our DB and if not, register it, right after the login, using the e-mail provided by the 3P `AdapterUser` (we also use the e-mail as a username in our DB as we didn't implement the option to let the user change its `username` yet).

Considering the above, there's no `username` property in `AdapterUser` to check against the users in the DB and even if this was true as soon as we implement the option for the user to change their username in PodCodar app, we wouldn't be able to check it against what comes from the 3P providers as our DB doesn't sync to 3P usernames.

Reply:
But this was useful as doing further research I realized I can use `User` instead of `AdapterUser`. But the issues are the same as when using it as `AdapterUser`.